### ISSUE TITLE
[Printing] Can't setup a Google Cloud Print printer

### ISSUE THREAD

**Issue by rdoursenaud** [Dolibarr/dolibarr#3235]
# Environment
- **Version**: 3.8.0-beta
- **OS**: Linux
- **Web server**: nginx/1.8.0
- **PHP**: fpm-fcgi 5.6.11
- **Database**: MySQL or MariaDB 5.5.5-10.0.20-MariaDB-log
- **URL**: /printing/admin/printing.php?mode=test&driver=printgcp
# Report

Can't setup a Google Cloud Print printer.
Login and password are set.

![capture d ecran de 2015-07-16 20-02-18](https://cloud.githubusercontent.com/assets/1210367/8731138/a9c1069a-2bf5-11e5-9b8b-8aaae48ddc89.png)

```
( ! ) Fatal error: Uncaught exception 'Exception' with message 'Please first login to Google by calling loginToGoogle function' in /home/rdoursenaud/Documents/gpcsrc/dolibarr/htdocs/core/modules/printing/printgcp.modules.php on line 139
( ! ) Exception: Please first login to Google by calling loginToGoogle function in /home/rdoursenaud/Documents/gpcsrc/dolibarr/htdocs/core/modules/printing/printgcp.modules.php on line 139
Call Stack
#   Time    Memory  Function    Location
1   0.0007  323664  {main}( )   .../printing.php:0
2   0.1057  10277672    printing_printgcp->listAvailablePrinters( ) .../printing.php:232
3   0.1058  10278048    printing_printgcp->getlist_available_printers( )    .../printgcp.modules.php:85
4   1.2287  10278480    printing_printgcp->get_printer_detail( )    .../printgcp.modules.php:125
```

**Comment by aspangaro** [comment]
Maybe @frederic34 can fix the problem ?

**Comment by frederic34** [comment]
Authentication is broken, I'm working on OAuht2

**Comment by rdoursenaud** [comment]
@frederic34 I already have a fairly good module handling OAuth2 authentication between Dolibarr & Google: https://github.com/GPCsolutions/zenfusionoauth
You should be able to leverage it easily if you don't want to reinvent the wheel.
Feel free to get in touch.

**Comment by frederic34** [comment]
@rdoursenaud I have already reinvented the wheel(s), thank's for your help
![2015-10-09 08_18_49-dolibarr - configuration](https://cloud.githubusercontent.com/assets/3624836/10387177/825689f2-6e5f-11e5-8fe7-8fe7717ad618.png)
![2015-10-09 08_19_49-dolibarr](https://cloud.githubusercontent.com/assets/3624836/10387179/847001dc-6e5f-11e5-8af7-5b41b729b472.png)
![2015-10-09 08_20_54-dolibarr](https://cloud.githubusercontent.com/assets/3624836/10387183/8912dc28-6e5f-11e5-94bc-910da60f7345.png)
![2015-10-09 08_24_59-dolibarr - reglages du module impression directe](https://cloud.githubusercontent.com/assets/3624836/10387184/8c5b9cda-6e5f-11e5-90c0-545991eb24c5.png)

**Comment by marcosgdf** [comment]
Should we move the module to experimental? Users are already complaining about it and it got reported more than 6 months ago

**Comment by frederic34** [comment]
It has been corrected in 3.9

**Comment by marcosgdf** [comment]
Then… when is it going to be merged to 3.8?

**Comment by eldy** [comment]
Won't be merged into 3.8. It wasn't a bug fix but a rewrite of a new implementation
